src/Syntax/TimeAgnostic.hs

```diff
@@ -106,21 +105,15 @@ deriving instance Eq   (InvariantPred t) => Eq   (Constructor t)
 
 data Behaviour t = Behaviour
   { _name :: Id
-  , _mode :: Mode
   , _contract :: Id
   , _interface :: Interface
-  , _preconditions :: [Exp ABoolean t]
+  , _preconditions :: [Exp ABoolean t] -- if preconditions are not satisfied execution is reverted
+  , _case :: [Exp ABoolean t] -- if preconditions are satisfied and a case is not, some other instance of the bahaviour should apply
```

Comment:
I'm fine with either name if you feel strongly, but do you think that something like `_assumptions` might be clearer here?